File: fix_octals.py
import re

with open('constants.ts', 'r') as f:
    content = f.read()

def replacer(match):
    args = match.group(1)
    # Replace leading zeros in numbers, but preserve single zero
    # e.g., 08 -> 8, 00 -> 0, 01 -> 1
    new_args = re.sub(r'\b0([0-9]+)\b', r'\1', args)
    return f"new Date({new_args})"

new_content = re.sub(r'new Date\(([^)]+)\)', replacer, content)

with open('constants.ts', 'w') as f:
    f.write(new_content)

File: fix_octals.js
const fs = require('fs');
let content = fs.readFileSync('constants.ts', 'utf8');

content = content.replace(/new Date\(([^)]+)\)/g, (match, args) => {
    const newArgs = args.split(',').map(arg => {
        let trimmed = arg.trim();
        // Replace leading zeros but not for a single zero
        if (/^0[0-9]+$/.test(trimmed)) {
            return trimmed.replace(/^0+/, '');
        }
        return trimmed;
    }).join(', ');
    return `new Date(${newArgs})`;
});

fs.writeFileSync('constants.ts', content);
console.log('Done.');
